gnovm/pkg/gnolang/type_check.go

```diff
@@ -260,24 +260,32 @@ Main:
 		ift := evalStaticTypeOf(store, last, currExpr.Func)
 		switch baseOf(ift).(type) {
 		case *FuncType:
-			tup := evalStaticTypeOfRaw(store, last, currExpr).(*tupleType)
-
 			// check for built-in functions
 			if cx, ok := currExpr.Func.(*ConstExpr); ok {
 				if fv, ok := cx.V.(*FuncValue); ok {
 					if fv.PkgPath == uversePkgPath {
 						// TODO: should support min, max, real, imag
-						switch {
-						case fv.Name == "len":
+						switch fv.Name {
+						case "len", "cap":
 							at := evalStaticTypeOf(store, last, currExpr.Args[0])
-							if _, ok := unwrapPointerType(baseOf(at)).(*ArrayType); ok {
-								// ok
-								break Main
+							exps := getExpr(currExpr.Args[0], &CallExpr{}, &seen[Expr]{})
+							for _, exp := range exps {
+								cx := exp.(*CallExpr)
```

Comment:
I don't understand how they are all CallExprs.

Reply:
They are all `CallExpr`s because the getExprs function is explicitly filtering for that type. When we call `getExpr`, we pass a `*CallExpr` as a template so that only nodes matching a call expression are collected.